### ISSUE TITLE
Error when overriding a property created via decorators

### ISSUE THREAD

**Issue by Cleptomania** [python/mypy#14684]
**Bug Report**

When a parent class has a property, created via decorators with a getter and setter. If you attempt to override the `fset` of the parent by using decorators in the subclass, mypy throws an error.

**To Reproduce**

```python
class A:
    def __init__(self, x: int):
        self._x = x

    @property
    def x(self) -> int:
        return self._x

    @x.setter
    def x(self, new_value: int):
        self._x = new_value


class B(A):
    def __init__(self):
        super().__init__(100)

    @A.x.setter
    def x(self, new_value: int):
        A.x.fset(self, new_value)

        print("Do something else ontop of the parent setter")
```

**Expected Behavior**

Mypy should produce no errors

**Actual Behavior**

Mypy produces the errors:

`overloaded function has no attribute "setter" [attr-defined]` on the decorator line in the subclass.
`overloaded function has no attribute "fset" [attr-defined]` on the call to fset in the subclass.

If instead of using a decorator, the property is created in the parent class via the `property()` function like below, then mypy does not produce any errors.

```python
class A:
    def __init__(self, x: int):
        self._x = x

    def _get_x(self) -> int:
        return self._x

    def _set_x(self, new_value: int):
        self._x = new_value

    x = property(_get_x, _set_x)
```

**Your Environment**

- Mypy version used: 1.0.0
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): Default configuration
- Python version used: 3.10.5

**Comment by tmke8** [comment]
Duplicate of #5936?

**Comment by Cleptomania** [comment]
It does seem to be the same problem, I couldn't find this exact problem when searching yesterday, if it makes sense feel free to close this in favor of the pre-existing one. I'll not the work-around I found to the problem in the other issue in case it proves helpful to anyone.
